### PR TITLE
Clean duplicate species

### PR DESCRIPTION
**Pull request by jwest60**
#111 

Added a command that deletes duplicates and merges any associated species and plants with the original species (clean:duplicate-species)

Tested and confirmed to delete and properly merge with the following cases:
- No duplicates exist
- One duplicate exists and isn't associated with anything
- One duplicate exists and is associated with a plant, tree and shrub species
- Site has a tree with duplicate species and shrub with original species

If you can think of any other cases, let me know and I'll test

### CODE REVIEW

File: app/Console/Commands/CleanDuplicateSpecies.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                $duplicates = $duplicates->orderBy('id', 'desc')->get();
+                $original = $duplicates->pop();
+                foreach ($duplicates as $duplicate) {
+                    $plants = $duplicate->plants()->cursor();
+                    foreach ($plants as $plant) {
+                        $plant->fill(['species_id' => $original->id])->save();
+                    }
+
+                    DB::table('site_species')->where('species_id', $duplicate->id)->update(['species_id' => $original->id]);
+                    DB::table('site_shrubs')->where('species_id', $duplicate->id)->update(['species_id' => $original->id]);
```
Comment (almasaeed2010) [inline app/Console/Commands/CleanDuplicateSpecies.php:56]: You need one more query for plants

File: app/Console/Commands/CleanDuplicateSpecies.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        $deleted = 0;
+
+        foreach (Species::cursor() as $species) {
+            $duplicates = Species::where('name', $species->name);
+            if ($duplicates->count() > 1) {
+                $duplicates = $duplicates->orderBy('id', 'desc')->get();
+                $original = $duplicates->pop();
+                foreach ($duplicates as $duplicate) {
+                    $plants = $duplicate->plants()->cursor();
```
Comment (almasaeed2010) [inline app/Console/Commands/CleanDuplicateSpecies.php:50]: There is no need to run through them one by one if you can use`DB::table...` instead

File: app/Console/Commands/CleanDuplicateSpecies.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    foreach ($plants as $plant) {
+                        $plant->fill(['species_id' => $original->id])->save();
+                    }
```
Comment (almasaeed2010) [inline app/Console/Commands/CleanDuplicateSpecies.php:53]: There is no need to run through them one by one if you can useDB::table... instead

**Comment by jwest60** [comment]
Changed plants query to
```
DB::table('plants')->where('species_id', $duplicate->id)->update(['species_id' => $original->id]);
```

**Pull request merged**